### PR TITLE
Add Latin locale hints and improve segmentation handling

### PR DESCRIPTION
**Pull request by dev-pi2pie**
Introduce optional Latin locale hints for ambiguous text to enhance locale detection.
Improve segmentation handling for mixed scripts and add documentation for non-word segment add-on and emoji/symbol segmentation. Include tests for various edge cases.

### CODE REVIEW

File: src/wc/locale-detect.ts [hunk 1]
```diff
@@ -72,6 +77,9 @@ export function detectLocaleForChar(
     if (hintedLocale !== DEFAULT_LOCALE) {
       return hintedLocale;
     }
+    if (options.latinLocaleHint) {
+      return options.latinLocaleHint;
+    }
     if (previousLocale && isLatinLocale(previousLocale)) {
```
[the remark this answers is not in the document]
Comment (dev-pi2pie) [reply src/wc/locale-detect.ts]: @codex address the feedback.
